lib/client-request.js

```diff
@@ -112,13 +112,14 @@ class RequestHandler extends AsyncResource {
   onComplete (trailers) {
     const { res } = this
 
-    removeSignal(this)
-
-    if (trailers) {
-      util.parseHeaders(trailers, this.trailers)
-    }
+    util.queueMicrotask(() => {
```

Comment:
why this?

Reply:
`onComplete` can be invoked before `await undici.request()` has continued execution. I think we can avoid the queueMicrotask before landing. For now this just gets tests to pass properly.

Reply:
fixed